src/lang/cpp/TypeSymbol.rsc

```diff
@@ -75,11 +75,12 @@ data TypeSymbol
   | \functionTemplate(loc decl, list[loc] templateParameters)
   | \variableTemplate(loc decl, list[loc] templateParameters)
   
-  | \aliasTemplate(loc decl, list[loc] templateParameters)
+  | \aliasTemplate(loc decl, list[loc] templateParameters, TypeSymbol \type)
   
   | \functionSetType(loc decl, list[TypeSymbol] templateArguments)
   | \functionSetTypePointer(loc decl, list[TypeSymbol] templateArguments)
   
+  | \unaryTypeTransformation(str operator, TypeSymbol operand)
```

Comment:
I'm not a big fan of encoding the operator as a string parameter. Can we specialize this per operator? Looking at the code, there's (currently) only one value in the operator enum anyway

Reply:
I don't like that either. I'll have a look. If it's a small finite set we can have one transformation constructor per operator. Otherwise, we introduce an TypeTransformationOperator sort.